lib/classes/PluginManager.js

```diff
@@ -148,7 +152,27 @@ class PluginManager {
     if (pluginsObject.localPath) {
       module.paths.unshift(pluginsObject.localPath);
     }
-    this.loadPlugins(pluginsObject.modules);
+    this.loadPlugins(pluginsObject.modules
+      .filter(name => name !== '@serverless/enterprise-plugin'));
+  }
+
+  loadEnterprisePlugin() {
+    const qualifiedEnterpriseModulePath = path.join(
+      __dirname, '../../node_modules/@serverless/enterprise-plugin');
+    // eslint-disable-next-line global-require
+    const sfePkgJson = require(`${qualifiedEnterpriseModulePath}/package.json`);
+    if (config.getGlobalConfig().enterpriseDisabled ||
+        !semver.satisfies(process.version, sfePkgJson.engines.node)) {
```

Comment:
the plugin & sfe have different node version requirements. this way allow sls to still support a slightly older version of node without requing the pugin to support them too

Reply:
Will it be the case once we drop support of Node.js v4 in SFO?

Have we considered to solve with `postinstall` script directly in `@serverless/enterprise-plugin` (?)

Reply:
> Will it be the case once we drop support of Node.js v4 in SFO?

yes, in fact node 4 has to be disabled because the upgraded npm doesn't work on it, not the plugin.

> Have we considered to solve with `postinstall` script directly in `@serverless/enterprise-plugin` (?)

Upgrading npm from a random package would violate user's trust IMO.

Reply:
This part of code doesn't seem to concern npm (at least to my understanding), 

Anyway I didn't read the intention right, this check is indeed harmless, just purpose seems questionable as:
- We start with situation where both SFO and SFE is supposed to support same versions of Node.js (v6+), so there's parity on that level.
- SFO will reference SFE via semver range, and SFE afaik will follow semver - therefore any drop of support of Node.js major within SFE, will be signaled by major version bump in SFE, and that will not affect existing SFO versions.

The use case it may serve, is when we will drop Node.js version support in SFE but don't want to do that in SFO while wanting to upgrade it to new SFE.

Doesn't seem clear to me why we may want to do that, and even if, wouldn't it be better to add such extra check then (so we avoid unnecessary noise now) ?